### PR TITLE
New design and new index page

### PR DESCRIPTION
**Pull request by okkebal**
The new index has 50 most popular questions and limits the amount of tags to 100. Lots of design changes.

### CODE REVIEW

File: sotoki/sotoki.py [hunk 1]
```diff
@@ -314,14 +315,33 @@ def startElement(self, name, attrs): #For each element
                 self.tags.append({'TagUrl': urllib.quote(attrs["TagName"]), 'TagName': attrs["TagName"], 'nb_post': int(attrs["Count"])})
 
     def endDocument(self):
-        jinja(
+	sql = "SELECT * FROM questiontag GROUP BY Tag ORDER BY Score DESC"
```
Comment (dattaz) [inline sotoki/sotoki.py]: this request isn't optimum, take hours on "big" dump

**Comment by dattaz** [comment]
General comment: 
- we use space for indentation
- templates_mini_mathjax hasn't been update
- we shouldn't have "stackexchange logo" because it's confusing ( for askubuntu or stackoverflow for exemple)
- title of questions on tags.html isn't vertically center
- on tags.html button "all tag" should be more different than tag
- tags.html can be rename index.html
- Border seem ok for me, but no background-color is better 

I've also made some specific comment directly on the code

If you made change i've point here i'll merge, put a message here when it's done :)
Thanks a lot !

**Comment by okkebal** [comment]
Ah, I discovered theres commenys here too. I have a new commit with your changes.

The background color is really nescessary for the design. Only borders makes for too much lines. By having a background color you can have no borders on most design elements, wich is much better!

**Comment by okkebal** [comment]
Added patch: removed logo, changed my editor to space indents and changed indentation in sotoki.py, alignment of title. I already changed the "All tags" link to be differnt from the other tags.

**Comment by okkebal** [comment]
So I added a commit called 'Changes requested by Dattaz' , but it does not appear here. Its in my fork. I guess you can see it and use it? Or do I have to do something?

**Comment by okkebal** [comment]
The Math thing: I changed the sotoki.py link to use all normal templates, but I forgot to link them back to the minified versions when I was done. In the end there is no change.. so a useless commit I guess. Sorry

**Comment by okkebal** [comment]
Would this be better?

SELECT * FROM questiontag GROUP BY Tag ORDER BY Score DESC LIMIT 50

By adding a limit will it make it faster, right?

**Comment by okkebal** [comment]
I made the SQL query very simple now: its also limited so should be very fast. Some removing of duplication is now done in Python. This results in the right list (the previous one was wrong).

I also removed the if-statement in the templates that removes tags there. Instead the array that is input is now trimmed.

**Comment by okkebal** [comment]
Oke, I'm done now. I kept seeing these small things in the design.

**Comment by dattaz** [comment]
Wow ! lot of new thing good !
Your now sql request is good !
I'll check that tommorow, but, i don't see any change on mathjax_templates and have you made something for https://github.com/openzim/sotoki/issues/53 ? (just to know, no obligation)

**Comment by okkebal** [comment]
Mathjax? Why are the same minified templates there too?

**Comment by okkebal** [comment]
The user badge has clickable picture and name, picture is much bigger, text is now aligned next to the picture.  is that #53?

**Comment by okkebal** [comment]
I've updated http://zorgsite.nl/files/ai.stackexchange.com_en_all_2017-06.zim you can take a look there at the latest version.

**Comment by dattaz** [comment]
because they include mathjs (just base.html change) (yep it's probally better to have a bool value in templates to include or not)
In your commit d4369c8 you remove this part "<script src="{{ rooturl }}/static/MathJax.js?config=TeX-AMS_HTML-full">"

**Comment by dattaz** [comment]
I've check your zim, it's seems really good for me. User badge is better!
One little more thing : Question score in index.hmtl and tag.html are not center (also not center in the current design) but if you can do something it's will more than perfect !

**Comment by kelson42** [comment]
It definitely has improved the last days, one thing which really bother me it the radius of the round corners, far to hight. 8px max.

**Comment by dattaz** [comment]
In templates/base.html should not be minify 
Something wrong with tag in responsive design (What you have done is also very good on mobile !)
capture d ecran 2017-06-09 a 09 24 03

Website using mathjax are in a mix beetwin old and new design.. anyway i have made a patch to remove mathjax templates and just have a bool value to include or not mathjax, i'll commit it after the merge

Except these details (+ comment of kelson) I agree to merge !

**Comment by okkebal** [comment]
in clean you should add "alltags.html"???

**Comment by okkebal** [comment]
I do not know what to do with a ".patch" file. Sorry

**Comment by okkebal** [comment]
How can I test this Mathjax thing, is there a page where this is used that I can check?? I think I just copied the minified versions into that folder.

**Comment by okkebal** [comment]
I do my tests by rendering ai.stackexchange.com every time, but I sure could use something that renders quicker. What do you use to check?

**Comment by dattaz** [comment]
So every thing is good for me, i'll merge. Is it ok for you ?

For mathjax you should try with a domain/site that use mathjax, for exemple meta.dsp.stackexchange.com. Anyway it's not a problem the 2 .patch will make that every work well.

**Comment by okkebal** [comment]
I added some last minute changes. If thats ok, please merge.

**Comment by dattaz** [comment]
I have merge and commit (39898f5358fbf3576dafb8d3e0f1b7dabbe58c46) change for mathjax.
Everything is good !
Thanks a lot for your contribution ! This is really appreciated !

**Pull request merged**